packages/compiler/src/constant_pool.ts

```diff
@@ -121,9 +121,45 @@ export class ConstantPool {
     if ((!newValue && !fixup.shared) || (newValue && forceShared)) {
       // Replace the expression with a variable
       const name = this.freshName();
-      this.statements.push(
-          o.variable(name).set(literal).toDeclStmt(o.INFERRED_TYPE, [o.StmtModifier.Final]));
-      fixup.fixup(o.variable(name));
+      let definition: o.DeclareVarStmt;
+      let usage: o.Expression;
+      if (isLongStringLiteral(literal)) {
+        // For string literals, Closure will **always** inline the string at
+        // **all** usages, duplicating it each time. For large strings, this
+        // unnecessarily bloats bundle size. To work around this restriction, we
+        // wrap the string in a function, and call that function for each usage.
+        // This tricks Closure into using inline logic for functions instead of
+        // string literals. Function calls are only inlined if the body is small
+        // enough to be worth it. By doing this, very large strings will be
+        // shared across multiple usages, rather than duplicating the string at
+        // each usage site.
+        //
+        // const myStr = function() { return "very very very long string"; };
+        // const usage1 = myStr();
+        // const usage2 = myStr();
+        definition =
+            o.variable(name)
+                .set(new o.FunctionExpr(
+                    [],  // Params.
+                    [
+                      // Statements.
+                      new o.CommentStmt(
+                          'Use a function to trick Closure to avoid duplicating this large string across mulitiple usages. See https://github.com/angular/angular/pull/38253.'),
+                      new o.ReturnStatement(literal),
```

Comment:
nit: for readability purposes may be split it into multiple lines (with string concat)?
(note: there is a small typo in the 'multiple')

```suggestion
                      new o.CommentStmt(
                          'Use a function to trick Closure to avoid duplicating this large string across multiple usages. ' +
                          'See https://github.com/angular/angular/pull/38253.'),
                      new o.ReturnStatement(literal),
```

Reply:
Split the comment up over multiple lines.